nornir/core/deserializer/configuration.py

```diff
@@ -0,0 +1,184 @@
+        jinja_filters = jinja_filter_func() if jinja_filter_func else {}
+        return configuration.Jinja2Config(filters=jinja_filters)
+
+
+class CoreConfig(BaseSettings):
+    num_workers: int = Schema(
+        default=20,
+        description="Number of Nornir worker threads that are run at the same time by default",
+    )
+    raise_on_error: bool = Schema(
+        default=False,
+        description=(
+            "If set to ``True``, (:obj:`nornir.core.Nornir.run`) method of "
+            "will raise an exception if at least a host failed"
+        ),
+    )
+
+    class Config:
+        env_prefix = "NORNIR_CORE_"
+        ignore_extra = False
+
+    @classmethod
+    def deserialize(self, **kwargs) -> configuration.CoreConfig:
+        c = CoreConfig(**kwargs)
+        return configuration.CoreConfig(**c.dict())
+
+
+class Config(BaseSettings):
+    core: CoreConfig = CoreConfig()
+    inventory: InventoryConfig = InventoryConfig()
+    ssh: SSHConfig = SSHConfig()
+    logging: LoggingConfig = LoggingConfig()
+    jinja2: Jinja2Config = Jinja2Config()
+    user_defined: Dict[str, Any] = Schema(
+        default={}, description="User-defined <k, v> pairs"
+    )
+
+    class Config:
+        env_prefix = "NORNIR_"
+        ignore_extra = False
+
+    @classmethod
+    def deserialize(cls, **kwargs) -> configuration.Config:
+        c = Config(
+            core=CoreConfig(**kwargs.pop("core", {})),
+            ssh=SSHConfig(**kwargs.pop("ssh", {})),
+            inventory=InventoryConfig(**kwargs.pop("inventory", {})),
+            logging=LoggingConfig(**kwargs.pop("logging", {})),
+            jinja2=Jinja2Config(**kwargs.pop("jinja2", {})),
+            **kwargs,
+        )
+        return configuration.Config(
+            core=CoreConfig.deserialize(**c.core.dict()),
+            inventory=InventoryConfig.deserialize(**c.inventory.dict()),
+            ssh=SSHConfig.deserialize(**c.ssh.dict()),
+            logging=LoggingConfig.deserialize(**c.logging.dict()),
+            jinja2=Jinja2Config.deserialize(**c.jinja2.dict()),
+            user_defined=c.user_defined,
+        )
+
+    @classmethod
+    def load_from_file(cls, config_file: str, **kwargs) -> configuration.Config:
+        config_dict: Dict[str, Any] = {}
+        if config_file:
+            yml = ruamel.yaml.YAML(typ="safe")
+            with open(config_file, "r") as f:
+                config_dict = yml.load(f) or {}
+        return Config.deserialize(**{**config_dict, **kwargs})
+
+
+def _resolve_import_from_string(import_path: Any) -> Optional[Callable[..., Any]]:
+    try:
+        if not import_path:
+            return None
+        elif callable(import_path):
+            return import_path
+        module_name = ".".join(import_path.split(".")[:-1])
+        obj_name = import_path.split(".")[-1]
+        module = importlib.import_module(module_name)
+        return getattr(module, obj_name)
```

Comment:
Some nit-picking, but I think this is a little bit more readable:
`module_name, obj_name = import_path.rsplit('.', 1)`

Reply:
done